features/collections.feature

```diff
@@ -263,6 +263,104 @@ Feature: Collections
     And the "_site/puppies/snowy.html" file should not exist
     And the "_site/puppies/hardy.html" file should not exist
 
+  Scenario: Access rendered collection with future date and unpublished flag via Liquid
+    Given I have a _puppies directory
+    And I have the following documents under the puppies collection:
+      | title  | date       | content             | published |
+      | Rover  | 2007-12-31 | content for Rover.  | true      |
+      | Figor  | 2007-12-31 | content for Figor.  | false     |
+      | Snowy  | 2199-12-31 | content for Snowy.  | true      |
+      | Hardy  | 2199-12-31 | content for Hardy.  | false     |
+    And I have a "_config.yml" file with content:
+    """
+    collections:
+      puppies:
+        output: true
+    """
+    And I have a "index.md" page that contains "{% for puppy in site.puppies %}<div>{{ puppy.title }}</div>{% endfor %}"
+    When I run jekyll build
+    Then I should get a zero exit status
+    And the _site directory should exist
+    And I should see "<div>Rover</div>" in "_site/index.html"
+    But I should see "<div>Snowy</div>" in "_site/index.html"
```

Comment:
A document with a future date gets published even without `--future` option?

Reply:
I don't think it gets _output_, but it does get _rendered_ and is accessible via Liquid. It's kind of weird behavior. 🤷‍♂️